docs/sources/auth/generic-oauth.md

```diff
@@ -54,6 +54,9 @@ Check for the presence of a role using the [JMESPath](http://jmespath.org/exampl
 
 See [JMESPath examples](#jmespath-examples) for more information.
 
+The `role_attribute_path` JMESPath expression will be applied first on the id_toke and userinfo by api_url, like all other options like `email_attribute_path`.
+If there is no hit it will look up the `access_token`.
```

Comment:
What do you mean by "no hit"?

Also, unclear antecedent. What is the "it" that will look up the access token? Please replace with a more specific noun if possible.

Reply:
Is the new sentence better?

The `role_attribute_path` JMESPath expression is first applied on the id_token and userinfo by api_url, like all other options. For example, `email_attribute_path`.
If there is no hit it will look up the `access_token`.	If there is no hit, then the JMESPath expression will be applied to the `access_token`.

Reply:
Still not sure what you mean by "no hit." Do you mean if the expression does not find a match?

Reply:
Yes that is what i mean. Sorry english is not my native language.

Reply:
@oddlittlebird Is this more clear?

The `role_attribute_path` JMESPath expression is first applied on the id_token and userinfo by api_url, like all other options. For example, `email_attribute_path`.
If the `role_attribute_path` JMESPath expression has no match in the id_toke or userinfo, then the expression will be applied to the `access_token`.

Reply:
Yes, much better!